### DISCUSSION TITLE
set a child element's style when the pointer is over its root element

### DISCUSSION THREAD

**Discussion by Kevinlan888** [AvaloniaUI/Avalonia#13603] [Q&A]
```
<Style Selector="Buttons|LinkButton">
	<Setter Property="Template">
		<ControlTemplate>
			<Grid Classes="wrapper" Background="Transparent">
				<ContentPresenter Content="{TemplateBinding Content}" />
				<Rectangle Classes="underline" Fill="{TemplateBinding Foreground}" Height="1"
						   VerticalAlignment="Bottom" Opacity="0" />
			</Grid>
		</ControlTemplate>
	</Setter>
</Style>
```

Hi, I got a problem when I tried to create a custom link button. is there any solution for setting rectangle's opacity when my mouse is over Grid. I can't set style on its child element when I selected Grid.

It's look like this:

```
<Style Selector="Grid.wrapper:pointerover">
	<Setter Property="Rectangle.Opacity" Value="1" />
</Style>
````

you would be my life saver if you can help me,  thanks.

**Comment by thevortexcloud** [comment]
Selectors work kind of like CSS selectors, so I assume this is what you want?

`Grid.wrapper:pointerover LinkButton`

**Reply by Kevinlan888** [reply]
Exactly, in fact, wpf can this too,  only need to name it before using it.
